calcom/cal.com
# fix: Dynamic events prevent fetching all users that share a username [CAL-2409]

## What does this PR do?



Currently there can be multiple usernames if they belong to different organizations. Right now in `handleNewBooking` all users of the same username were being fetched in dynamic bookings, without any context to the org or not. The PR only fetches users that belong to the org or just normal users.

For example we have two users with the "admin" username "admin@example.com" and "org1member1@org1.com". We can see when we create a dynamic event with "admin+free" before it was grabbing both users but now it is querying for one. This works within an orgs context as well

## Before
![image](https://github.com/calcom/cal.com/assets/65426560/8d92afd2-cf1c-47b3-a0a6-9cd8e1533959)

## After
![image](https://github.com/calcom/cal.com/assets/65426560/9717c725-dcac-43c8-9db1-16fb98a7043d)



Fixes #11186 



## Requirement/Documentation



- If there is a requirement document, please, share it here.
- If there is ab UI/UX design document, please, share it here.

## Type of change





## How should this be tested?



- Create two users with the same username. The second user should belong to an org
- Create a dynamic booking that includes the test username
- The created booking should only contain the user outside of the booking
- Go through this flow again but within an org context

## Mandatory Tasks


## Checklist



- I haven't added tests that prove my fix is effective or that my feature works
- I haven't checked if new and existing unit tests pass locally with my changes
